homework-assignments-1/task4/task4.ts

```diff
@@ -0,0 +1,169 @@
+class Product {
+    constructor(
+        public name: string,
+        public price: number,
+        public stock: number
+    ) {}
+
+    addStock(quantity: number) {
+        this.stock += quantity;
+        return this.stock;
+    }
+
+    reduceStock(quantity: number) {
+        this.stock = Math.max(this.stock - quantity, 0);
+        return this.stock;
+    }
+}
+
+class Customer {
+    constructor(
+        public name: string,
+        public email: string,
+        public cart: Product[] = [],
+        public address: string
+    ) {}
+
+    addToCart(product: Product) {
+        this.cart.push(product);
```

Comment:
check before is there any stock available in market. if (product.getStock() > 0)